src/Utilities/AnsiBuilder.cs

```diff
@@ -0,0 +1,330 @@
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+
+namespace WheelMUD.Utilities
+{
+    /// <summary>
+    /// Mutable String class for Ansi based strings, optimized for speed and memory while retrieving the final result
+    /// as a string. Similar use to StringBuilder, but avoid a lot of allocations done by StringBuilder
+    /// </summary>
+    public class AnsiBuilder
+    {
+        /// <summary>
+        ///     Working mutable string
+        /// </summary>
+        private char[] buffer;
+
+        private int bufferPos;
+        private int charsCapacity;
+
+        /// <summary>
+        /// Temporary string used for the Replace method
+        /// </summary>
+        private List<char> replacement;
+
+        /// <summary>
+        /// Immutable string. Generated at last moment, only if needed
+        /// </summary>
+        private string stringGenerated = "";
+
+        /// <summary>
+        /// Dirty flag for AnsiBuilder containing ansi that needs parsing
+        /// </summary>
+        private bool needsParsing;
+
+        public AnsiBuilder(int initialCapacity = 32)
+        {
+            buffer = new char[charsCapacity = initialCapacity];
+        }
+
+        public int Length => ToString().Length;
+
+        /// <summary>
+        /// Return the string
+        /// </summary>
+        public override string ToString()
+        {
+            if (needsParsing) Parse();
```

Comment:
nit: The vast majority of the code base uses explicit `{ }` scopes on all conditionals, for loops, and so on even when trivial.  However, per #24 as you discovered already, we can figure this stuff out later.